### PR TITLE
Prevent double logging of the airflow container logs

### LINKED ISSUE #473: Logs from the airflow container are logged twice
Logs from the airflow container are logged twice.

They are correctly written with the logging framework to the file `airflow.py.json`:

```json
{
  "cluster": "airflow",
  "container": "airflow",
  "file": "airflow.py.json",
  "level": "INFO",
  "logger": "flask_appbuilder.baseviews",
  "message": "Registering route / ('GET',)",
  "namespace": "kuttl-test-heroic-badger",
  "pod": "airflow-scheduler-default-0",
  "role": "scheduler",
  "roleGroup": "default",
  "timestamp": "2024-07-15T17:53:18.924Z"
}
```

Additionally, the standard output is incorrectly captured (including the ANSI escape codes) and written to the file `container.stdout.log`:

```json
{
  "cluster": "airflow",
  "container": "airflow",
  "file": "container.stdout.log",
  "level": "INFO",
  "logger": "ROOT",
  "message": "[\u001b[34m2024-07-15T17:53:18.924+0000\u001b[0m] {\u001b[34mbaseviews.py:\u001b[0m319} INFO\u001b[0m - Registering route / ('GET',)\u001b[0m",
  "namespace": "kuttl-test-heroic-badger",
  "pod": "airflow-scheduler-default-0",
  "role": "scheduler",
  "roleGroup": "default",
  "timestamp": "2024-07-15T17:53:19.882505453Z"
}
```

This problem was introduced in #311 in [airflow_controller.rs#L748-L752](https://github.com/stackabletech/airflow-operator/blob/8164085d00f54c12c8fec0ba71f7b03619e61e67/rust/operator-binary/src/airflow_controller.rs#L748-L752).

The shell output must not be captured anymore but it must also be ensured that the logs of the Kubernetes executors are still gathered.

### PR DESCRIPTION
**Pull request by siegfriedweber**
# Description

Prevent double logging of the airflow container logs

Both executors are now tested in the logging test.

Fixes #473 

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by siegfriedweber** [comment]
https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/2/

**Review by NickLarsenNZ — DISMISSED** [review]
LGTM, although I am not so familiar with the aggregator rules.

File: tests/templates/kuttl/commons/metrics.py [hunk 1]
```diff
@@ -12,18 +12,23 @@ def exception_handler(exception_type, exception, traceback):
...
     assert metric_response.status_code == 200, \
         f"Metrics could not be retrieved from the {role}."
     return metric in metric_response.text
 
 
+try:
+    role_group = sys.argv[1]
+except IndexError:
+    role_group = "default"
+
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply tests/templates/kuttl/commons/metrics.py:26]: Oops, meant to delete this one (they are standalone scripts, so it doesn't make sense).

File: rust/operator-binary/src/airflow_controller.rs [hunk 2]
```diff
@@ -824,19 +823,6 @@ fn build_server_rolegroup_statefulset(
...
 
     let mut airflow_container_args = Vec::new();
-    if let Some(ContainerLogConfig {
-        choice: Some(ContainerLogConfigChoice::Automatic(log_config)),
-    }) = merged_airflow_config
-        .logging
-        .containers
-        .get(&Container::Airflow)
-    {
-        airflow_container_args.push(capture_shell_output(
```
[the remark this answers is not in the document]
Comment (siegfriedweber) [reply rust/operator-binary/src/airflow_controller.rs:834]: The airflow-operator configures Airflow to log into the file `airflow.py.json`:
https://github.com/stackabletech/airflow-operator/blob/3c7769896dcdbdcbe23942d9128afc36212d37a8/rust/operator-binary/src/product_logging.rs#L165-L172

The operator also deploys the Vector agent side-car container:
https://github.com/stackabletech/airflow-operator/blob/3c7769896dcdbdcbe23942d9128afc36212d37a8/rust/operator-binary/src/airflow_controller.rs#L1022-L1033

The Vector container is configured to watch `*.py.json` files:
https://github.com/stackabletech/operator-rs/blob/stackable-operator-0.70.0/crates/stackable-operator/src/product_logging/framework.rs#L745-L748

**Comment by siegfriedweber** [comment]
https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/4/

**Comment by siegfriedweber** [comment]
:hourglass_flowing_sand: https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/5/

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**